components/elm/src/biogeochem/AllocationMod.F90

```diff
@@ -879,15 +897,45 @@ subroutine TotalNPDemand(num_soilp,filter_soilp, photosyns_vars, &
                g1 = 0.25_r8
 
             else if (croplive(p) .and. percrop(ivt) == 1.0_r8) then
-               arepr(p) = 0._r8
-               aroot(p) = max(0._r8, min(1._r8, arooti(ivt) -   &
-                    (arooti(ivt) - arootf(ivt)) *  &
-                    min(1._r8, hui(p)/gddmaturity(p))))
-               fleaf = fleafi(ivt) * (exp(-bfact(ivt)) -         &
-                    exp(-bfact(ivt)*hui(p)/gddmaturity(p))) / &      ! replacing huigrain with gddmaturity since huigrain does not exist for perennial crops
-                    (exp(-bfact(ivt))-1) ! fraction alloc to leaf (from J Norman alloc curve)
-               aleaf(p) = max(1.e-5_r8, (1._r8 - aroot(p)) * fleaf)
-               astem(p) = 1._r8 - arepr(p) - aleaf(p) - aroot(p)
+               if (ivt==nsugarcane .or. ivt==nsugarcaneirrig) then
+
+                  rmat = 100.0_r8 * (onset_gdd(p)/gddmaturity(p))
+
+                  if (nyrs_crop_active(p) == 0) then ! Year 1
+                     aerial(p) = (1.0_r8 - arootf(ivt)) * min(1.0_r8, (1 - exp(-(rootd * 0.6 * rmat))))
```

Comment:
What the source or logic behind these seasonal curves? An explanatory comment might be useful here.

Reply:
> What the source or logic behind these seasonal curves? An explanatory comment might be useful here.

@bpbond I have added reasoning for carbon allocation to various plant components.